### PR TITLE
feat: support single item book spine

### PR DESCRIPTION
**Pull request by eugenserbanescu**
Does what it says on the tin :)

### CODE REVIEW

**Review by RicheyRyan — APPROVED** [review]

**Review by neil-buckley — APPROVED** [review]
Thanks @eugenserbanescu - been a long standing issue!

File: extract/spine.js [hunk 1]
```diff
@@ -21,9 +21,11 @@ const PROPERTIES = {
 export default function spine(xml, tocItem=false) {
   // TODO This will need to change when we use the spine to navgate the book
   // and allow hidden toc items
-  const items = tocItem ?
-    xml.package.spine.itemref.filter(i => i.id !== tocItem) :
-    xml.package.spine.itemref;
+  let items = Array.isArray(xml.package.spine.itemref) ? xml.package.spine.itemref : [xml.package.spine.itemref];
+
+  if (tocItem) {
+    items = items.filter(i => i.id !== tocItem);
```
Comment (dariocravero) [inline extract/spine.js:27]: If `xml.package.spine.itemref` isn't an array, would `tocItem` be always false in then?
If so, could we put the filter on line 24 and avoid the if altogether?

Comment (eugenserbanescu) [reply extract/spine.js:27]: `tocItem` won't necessarily be false. A book may have a toc even if it has a single page in it, right? I'll look at the spec to try and get a definitive answer

Comment (eugenserbanescu) [reply extract/spine.js:27]: Apparently epubs need to have a toc: "The toc nav is the only navigation aid that has to be included in the EPUB Navigation Document", therefore that `tocItem` won't ever be falsey. We could potentially get rid of the if altogether and just force the spine to be an array and filter it anyway. Thoughts? @dariocravero

Comment (dariocravero) [reply extract/spine.js:27]: No worries, leave it as it is for now

**Review by dariocravero — APPROVED** [review]

**Comment by dariocravero** [comment]
@eugenserbanescu published as v4.2.3

**Pull request merged**